File: 0409-longest-palindrome/0409-longest-palindrome.java
class Solution {
    public int longestPalindrome(String s) {
        Map<Character, Integer> map = new HashMap<>();
        int answer = 0;
        
        for(char c : s.toCharArray()) {
            map.put(c, map.getOrDefault(c, 0) + 1);
            if(map.get(c) % 2 == 0) {
                answer += map.get(c);
                map.put(c, map.get(c) % 2);
            }
        }
        
        return answer < s.length() ? answer + 1 : answer;
    }
}

File: 0409-longest-palindrome/0409-longest-palindrome.cpp
class Solution {
public:
    int longestPalindrome(string s) {
        map<char, int> cmap;
        for(auto& c : s) {
            if(!cmap.contains(c)) {
                cmap.emplace(c, 0);
            }
            cmap[c]++;
        }
        
        int answer = 0;
        for(auto& m : cmap) {
            answer += (m.second - m.second % 2);
            
        }
        return answer < s.size() ? answer + 1 : answer;
    }
};